File: 2022.01.31/javascript/15829.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n')[1];
let result = 0;
for (let i = 0 ; i<input.length ; i++){
  char = input[i];
  result += ( (char.charCodeAt() - "a".charCodeAt() + 1) % 124567891 * (31**i)% 1234567891 );
}
console.log(result%1234567891);

File: Implementation/python/15829.py
n = int(input())
s = input()
result = 0
for i in range(len(s)):
    char = s[i]
    result += ( (ord(char) - ord("a") + 1) % 124567891 * (31**i)% 1234567891 )
print(result%1234567891)